commit 48058290182a60e9d13d082b2bce89d8b02af0da (refs/changes/08/4608/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-07-31T20:32:14+00:00 (1 year, 2 months ago)
    
    joyent/node-sshpk#52 Buffer no longer performs length check for hex strings in node v8.x

diff --git a/lib/fingerprint.js b/lib/fingerprint.js
index 4e75e66..62f3754 100644
--- a/lib/fingerprint.js
+++ b/lib/fingerprint.js
@@ -102,7 +102,7 @@ Fingerprint.parse = function (fp, options) {
 		parts = parts.join('');
 		/*JSSTYLED*/
 		var md5RE = /^[a-fA-F0-9]+$/;
-		if (!md5RE.test(parts))
+		if (!md5RE.test(parts) || md5RE.length % 2 !== 0)
 			throw (new FingerprintFormatError(fp));
 		try {
 			hash = Buffer.from(parts, 'hex');
diff --git a/test/openssl-cmd.js b/test/openssl-cmd.js
index a78c795..d1472b6 100644
--- a/test/openssl-cmd.js
+++ b/test/openssl-cmd.js
@@ -212,7 +212,12 @@ function genTests() {
 		var bufs = [];
 		kid.stdout.on('data', bufs.push.bind(bufs));
 		kid.on('close', function (rc) {
-			t.equal(rc, 0);
+			t.equal(rc, 0, 'openssl exited 0');
+			if (bufs.length === 0) {
+				t.fail('"openssl dgst" wrote no output');
+				t.end();
+				return;
+			}
 			var output = Buffer.concat(bufs);
 			var sig = sshpk.parseSignature(output, algo, 'asn1');
 			t.ok(sig);
